Added database to the project.

AzizDeveloper opened the pull request:
Created Movie, MovieMapper, MovieRepository, liquibase.propertiesm db.changelog-1.0.sql, db.changelog-master.xml, db.data-1.0.sql, MovieMapperTest, MovieRepositoryTest, application-test.yml, test-data.sql, test-schema.sql.
Added spring-boot-data-jpa, postgresql, mapstruct, mapstruct-processor, dbunit, liquibase, h2 dependencies and docker-compose.yml file.

AzizDeveloper committed:
Created Movie, MovieMapper, MovieRepository, liquibase.propertiesm db.changelog-1.0.sql, db.changelog-master.xml, db.data-1.0.sql, MovieMapperTest, MovieRepositoryTest, application-test.yml, test-data.sql, test-schema.sql.
Added spring-boot-data-jpa, postgresql, mapstruct, mapstruct-processor, dbunit, liquibase, h2 dependencies and docker-compose.yml file.

serlesen commented on src/main/java/dev/aziz/movies/controllers/MovieController.java:
```diff
@@ -37,8 +37,8 @@ public ResponseEntity<MovieDto> getMovie(@PathVariable Long id) {
     @PostMapping("/movies")
-    public ResponseEntity<MovieDto> createMovie(@RequestBody @Valid MovieDto movieDto) {
-        return ResponseEntity.created(URI.create("/movies/" + movieDto.getId())).body(movieService.createMovie(movieDto));
+    public ResponseEntity<MovieDto> createMovie(@RequestBody @Valid Movie movie) {
```
Don't use the entities in the controllers, use the DTOs and use a mapper in the Service to transform an entity into a DTO.

serlesen commented on src/main/resources/sql/db.changelog-1.0.sql:
```diff
@@ -0,0 +1,15 @@
+    id          BIGSERIAL PRIMARY KEY,
+    title       TEXT    NOT NULL CHECK (title <> ''),
+    description TEXT    NOT NULL CHECK (description <> ''),
+    produced_year INTEGER NOT NULL CHECK (year BETWEEN 1850 AND 2050),
```
The check is done with a column named `year`, but the column is created as `produced_year`.

serlesen commented on src/main/java/dev/aziz/movies/entities/Movie.java:
```diff
@@ -0,0 +1,63 @@
+    private int rate;
+
+    @NotEmpty(message = "Director should not be empty")
+    private String director;
```
You should have another table with all the directors. And have a foreign key to the movies table.

serlesen commented on src/main/java/dev/aziz/movies/entities/Movie.java:
```diff
@@ -0,0 +1,63 @@
+    private String director;
+
+    @NotEmpty(message = "Main actors list should not be empty")
+    private String mainActors;
```
You should have another table with all the actors. And have a foreign key to the movies table.

serlesen commented on src/main/java/dev/aziz/movies/entities/Movie.java:
```diff
@@ -0,0 +1,63 @@
+    private String mainActors;
+
+    @NotEmpty(message = "Genre should not be empty")
+    private String genre;
```
You should have another table with all the genres. And have a foreign key to the movies table.

serlesen commented on src/test/java/dev/aziz/movies/MoviesApplicationTests.java:
```diff
@@ -6,8 +6,8 @@
-    @Test
-    void contextLoads() {
-    }
+//    @Test
```
Why did you comment this test?

AzizDeveloper committed:
Created new entities and integrated to the project.

Created Actor, Director, Genre entities and repositories. Created relationships with these entities and movie. Updated sql files. Updated tests.

serlesen commented on src/main/java/dev/aziz/movies/services/MovieService.java:
```diff
@@ -73,19 +66,45 @@ public MovieDto updateById(Long id, UpdateMovieDto updateMovieDto) {
-            movieById.setGenre(updateMovieDto.getGenre());
+        if (updateMovieDto.getGenres() != null) {
+            List<Genre> genreList = new ArrayList<>();
+            for (int i = 0; i < updateMovieDto.getGenres().size(); i++) {
```
Instead of a for loop, can you use a for-each loop? Or even better with a Lambda?

AzizDeveloper committed:
For loops in MovieService class changed to lambda expressions
